tests/Unit/Core/Form/IdentifiableObject/CommandBuilder/Product/ProductOptionsCommandsBuilderTest.php

```diff
@@ -164,4 +181,102 @@ public function getExpectedCommands()
             [$command],
         ];
     }
+
+    /**
+     * @return Generator
+     */
+    public function getExpectedCommandsForAllShops(): Generator
+    {
+        $command = $this->getAllShopsCommand();
+        $command->setCondition(ProductCondition::NEW);
+        yield [
+            [
+                'specifications' => [
+                    'condition' => 'new',
+                    self::MODIFY_ALL_NAME_PREFIX . 'condition' => true,
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setShowCondition(false);
+        yield [
+            [
+                'specifications' => [
+                    'not_handled' => 0,
+                    'show_condition' => 0,
+                    self::MODIFY_ALL_NAME_PREFIX . 'show_condition' => true,
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setShowPrice(false);
+        yield [
+            [
+                'options' => [
+                    'not_handled' => 0,
+                    'visibility' => [
+                        'show_price' => false,
+                        self::MODIFY_ALL_NAME_PREFIX . 'show_price' => true,
+                    ],
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setAvailableForOrder(true);
+        yield [
+            [
+                'options' => [
+                    'visibility' => [
+                        'available_for_order' => '1',
+                        self::MODIFY_ALL_NAME_PREFIX . 'available_for_order' => true,
+                    ],
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setVisibility(ProductVisibility::INVISIBLE);
+        yield [
+            [
+                'options' => [
+                    'visibility' => [
+                        'visibility' => ProductVisibility::INVISIBLE,
+                        self::MODIFY_ALL_NAME_PREFIX . 'visibility' => true,
+                    ],
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setOnlineOnly(false);
+        yield [
+            [
+                'options' => [
+                    'visibility' => [
+                        'online_only' => false,
+                        self::MODIFY_ALL_NAME_PREFIX . 'online_only' => true,
+                    ],
+                ],
+            ],
+            [$command],
+        ];
+    }
```

Comment:
You should also add a few use cases with both single shop command and all shops command